### ISSUE TITLE
hx-get does not execute <script src=""></script>

### ISSUE THREAD

**Issue by BoPeng** [bigskysoftware/htmx#379]
I was puzzled why some of my scripts obtained by `hx-get` got executed, and some did not. Reading

https://github.com/bigskysoftware/htmx/blob/b8ffa5c6a5176ac452e5aeee2710a2456dcde8d1/dist/htmx.js#L1257-L1267

I realized that `hx-get` only execute scripts embedded inside `<script></script>`, not specified with `<script src="/path/to/script"></script>`. Is there any reason why it is designed this way?

**Comment by BoPeng** [comment]
Another case I encountered is that when the content loaded by `hx-get` has some JS code that makes another AJAX call to a separate site, it would fail with a 500 cors error.

**Comment by benpate** [comment]
Regarding `<script src="/path/to/script">` tags, I'd expect that they would simply work as requested when the new HTML is added into the DOM.  But, I haven't checked this to be sure, so I'll have to go with your experimental results.  It sure seems like they *should* work, whether it's an oversight in htmx or not.  As a workaround, is it something that *must* be loaded dynamically via htmx, or could you simply require this script on the main page when it first loads?

Regarding the "CORS" error, that's not related to htmx.  That's a configuration issue between your site and the remote site.  The remote site may have specifically limited their scripts from running on any domain that they don't control.  CORS is a big mess of issues, and can take some work to get around, but it's not a bug in htmx.  The specifics will depend on the remote site you're accessing, but I'd start by Googling "CORS Error" and going from there.

**Comment by BoPeng** [comment]
> I'd expect that they would simply work as requested when the new HTML is added into the DOM.

I am not sure about this part. If that is the case, why the `<script>code</script>` case cannot be handled by the DOM directly and have to be executed by htmx? I can look further into the issue (e.g. write a test case) if there is no reason for HTMX not to execute `<script src=""></script>`.

**Comment by 1cg** [comment]
I don't believe adding a script tag within the content of the body and a `src` attibute is allowed.  I know that script tags in general are not interpreted when added to the body, and we had to add special case handling for this situation.

**Comment by BoPeng** [comment]
According to [W3 school](https://www.w3schools.com/js/js_whereto.asp#:~:text=JavaScript%20in%20or%20%3C,HTML%20page%2C%20or%20in%20both)

> You can place an external script reference in <head> or <body> as you like.
> The script will behave as if it was located exactly where the <script> tag is located.

In my particular application, I have some django forms with JS code and everything works when the forms are loaded as complete pages with `<script></script>` inside `<body>` (after form `<div>`). When the forms are loaded via `hx-get`, I was glad to find some forms worked but some others did not. It turned out `<script>code</script>` worked and `<script src=""></script>` did not.

I donot know how difficult it is to support `<script src=""><script>` but this should at least be documented.

**Comment by BoPeng** [comment]
After experimenting with a few methods to get around of this problem, I ended up using something like the following

```
<html lang="en">
<body>
  <a href="#" hx-get="load.html" hx-trigger="load, click">
  </a>

  <script src="https://unpkg.com/htmx.org@1.2.1"></script>
  <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
<script>
document.addEventListener('htmx:afterSwap', function() {
   $('.myelement').whatever();
})
</script>
</body>

</html>
```

Basically,

1. I put all the `<script src=""></script>` in the master `html` file so there is no need to execute them after `hx-get`.
2. I put all the process functions inside `htmx:afterSwap`, also in the master `html` file.

In this way, there is no need to include `JS` code with content loaded from `hx-get`. When `hx-get` is completed, all JS code is already in the master `html` file and will be triggered by `hxmx:afterSwap`.  I could also use self-defined events if a general `htmx;afterSwap` event does not cut it. The only problem is that the master html file will be slow to load, but cannot be as bad as a SPA page.

**Comment by 1cg** [comment]
OK, sorry for the long delay on this.

Here is the fundamental problem: script tags do not execute when you insert them into the DOM via a javascript API, so it's up to you to do so.  We do this with inline scripts, in kind of a hacky way.  

Script tags w/ `src` attributes require us to fetch the script, however, and the semantics unfortunately can't be the same as "normal" HTML.  With normal HTML, a script tag will be completely downloaded and executed before the rest of the DOM is processed, IIRC.  Unfortunately we can't do that.  The best we could do would be to load the scripts up asynchronously, which could have strange effects on evaluation.  This is why I have always punted on this and recommended loading your javascript up when the inital file loads.

I can think more about the issue.

**Comment by BoPeng** [comment]
Thanks @1cg . I understand the technical side of the issue and the inline script works well for me. I have figured out and used a [solution ](https://github.com/bigskysoftware/htmx/issues/379#issuecomment-794671525) that works reasonably well for my web site, with the only caveat being the big `afterSwap` function that will be used to handle all possible after-swap actions from all pages, which can be annoying but at least it works and the burden is only the client side.

**Comment by delitescere** [comment]
This same scenario (progressive enhancement loading a fragment of another page which includes a `<script>`) works fine with Hotwire and it's Turbo Frames. I wonder what is being done differently?

**Comment by 1cg** [comment]
fix ready for next release: https://github.com/bigskysoftware/htmx/commit/60a67179941b98132ac577abf1b42ea030231b65

**Comment by StitiFatah** [comment]
> fix ready for next release: [60a6717](https://github.com/bigskysoftware/htmx/commit/60a67179941b98132ac577abf1b42ea030231b65)

Is this expected that scripts with src and type="module" only execute on the first one but not on subsequent swaps ? Let me explain :

If I I have two endpoints, /foo and /bar, with `<script  src="/foo.js" type="module"> ` and `<script  src="/bar.js" type="module"> `  if I manually navigate to /foo then foo.js will execute,  if I swap the html by the the one returned by /bar then bar.js will execute, now if I swap the html by the one returned by /foo foo.js will NOT  execute, same if we continue the back and forth, foo.js and bar.js will only execute once. 

This behavior doesn't occur when working with non type="module" and inline scripts where the javascript will execute for each swap. 

Why does it behave that way and is there a workaround ?
